x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_search_application_put.yml

```diff
@@ -109,29 +115,34 @@ teardown:
         body:
           indices: [ "test-index1", "test-index2" ]
           template:
-            script: {
-              source: "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"_all\" } } }",
-              params: {
+            script:
+              source: "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"_all\" } } }"
+              params:
                 query_string: "elastic"
-              },
               lang: "mustache"
-            }
+            dictionary:
+              properties:
+                another_query_string:
```

Comment:
Do we want this request to succeed? Even though it's string encoded, this template is specifying the parameter `query_string` and it's not in the dictionary.

Reply:
The dictionary does not include the parameters in the template. There are the following reasons for that:
- Default params are included as part of the template creation, so they can be validated by the user that updates the template 
- It's not enough to validate the default params via a dictionary, but they need to be checked with the template as well to ensure it is rendered properly.
- It's easier to implement mandatory params for a template this way, as it's enough to include `required` params in the dictionary. Otherwise, we need to provide a separate mechanism for validating mandatory parameters.

I'm happy to discuss this - for now I think it's a good compromise and allows for the most flexibility in _search params validation. I know @jimczi has thoughts on this as well, so let's discuss - hopefully in a separate PR 😅

Reply:
I'm OK with this for the PR, happy to discuss follow up iterations 👍